src/storage/hnsw_usearch.cpp

```diff
@@ -0,0 +1,500 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+//
+// hnsw_usearch.cpp - implementation of HNSW index using USearch
+//
+
+#include "hnsw.hpp"
+#include "error_manager.h"
+#include "system_parameter.h"
+#include "vector_opfunc.hpp"
+#include "boot_sr.h"
+#include "file_io.h"
+#include "system_parameter.h"
+#include "dbtype.h"
+#include "porting.h"
+#include "vector_distance_enum.h"
+#include <fstream>
+#include <filesystem>
+
+#include "usearch/index.hpp"
+#include "usearch/index_dense.hpp"
+#include "usearch/index_plugins.hpp"
+
+// XXX: SHOULD BE THE LAST INCLUDE HEADER
+#include "memory_wrapper.hpp"
+
+
+#include "strict_warnings_on.hpp"// TODO : When cub_server terminates, hnsw_index_id will be reset to 0.
+//        This is not a problem in current implementation, but it may be a problem in the future,
+//        such as duplicate hnsw_index_id when cub_server restarts.
+//        We need to consider a better way to identify the hnsw index.
+
+
+using namespace unum;
+
+int hnsw_index_id = 0;
+std::unordered_map<int, std::unique_ptr<usearch::index_dense_t>> hnsw_index_map;
+char hnsw_index_directory[PATH_MAX] = {0};
+bool hnsw_index_directory_created = false;
+static std::mutex hnsw_elem_mutex;
+
+static int dump_hnsw_index (int hnsw_id, const std::unique_ptr<usearch::index_dense_t> &index);
+static int get_hnsw_index_file_path (int hnsw_id, char *out_path);
+static int create_hnsw_index_directory ();
+static bool is_hnsw_index_file_exists (int hnsw_id);
+static int load_hnsw_index_from_file (int hnsw_id);
+static int hnsw_check_and_load_index (int hnsw_id);
+static void get_new_hnsw_index_id ();
+static int64_t encode_oid (const OID &oid);
+static OID decode_oid (int64_t encoded_oid);
+
+BTID *
+xhnsw_add_index (THREAD_ENTRY *thread_p, BTID *btid, int dimension = 10, int hnsw_M = 16, int hnsw_efConstruction = 64,
+		 int metric = DB_VECTOR_DISTANCE_METRIC::METRIC_EUCLIDEAN)
+{
+  usearch::metric_kind_t metric_kind = usearch::metric_kind_t::unknown_k;
+  switch (metric)
+    {
+    case METRIC_UNKNOWN:
```

Comment:
발생할 수 없는 시나리오라면, 다음 코드 추가
```suggestion
    case METRIC_UNKNOWN:
      ASSERT_CUBVEC (false);
```
```